File: 3208_alternating_groups_ii/solution.go
// LeetCode 3208 - Alternating Groups II
// https://leetcode.com/problems/alternating-groups-ii/

func numberOfAlternatingGroups(colors []int, k int) (ans int) {
	n := len(colors)
	cnt := 0
	for i := 0; i < n<<1; i++ {
		if i > 0 && colors[i%n] == colors[(i-1)%n] {
			cnt = 1
		} else {
			cnt++
		}
		if i >= n && cnt >= k {
			ans++
		}
	}
	return
}

File: 3208_alternating_groups_ii/solution.php
<?php
// LeetCode 3208 - Alternating Groups II
// https://leetcode.com/problems/alternating-groups-ii/

class Solution {
    function numberOfAlternatingGroups($colors, $k) {
        $n = count($colors);
        $cnt = 0;
        $ans = 0;
        for ($i = 0; $i < $n * 2; $i++) {
            if ($i > 0 && $colors[$i % $n] === $colors[($i - 1) % $n]) $cnt = 1;
            else $cnt++;
            if ($i >= $n && $cnt >= $k) $ans++;
        }
        return $ans;
    }
}
